### PR TITLE
DAT-2895 Support gallery as image source in new infoboxes

### PR DESCRIPTION
**Pull request by sebastianmarzjan**

### CODE REVIEW

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php [hunk 1]
```diff
@@ -44,15 +48,45 @@ private function getFilePrefixRegex( $contLang ) {
 	 * @return mixed
 	 */
 	public function sanitizeImageFileName( $filename, $contLang ) {
-		// replace the MW square brackets and surrounding whitespace
-		$trimmedFilename = trim( $filename, "\t\n\r[]" );
-
+		$plainText = $this->convertToPlainText( $filename );
 		$filePrefixRegex = $this->getFilePrefixRegex( $contLang );
+		$textLines = explode( PHP_EOL, $plainText );
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: does every image in gallery is always separated by new line?

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php [hunk 2]
```diff
@@ -44,15 +48,45 @@ private function getFilePrefixRegex( $contLang ) {
...
+
+	/**
+	 * @param $filename
+	 * @return string
+	 */
+	private function convertToPlainText( $filename ) {
+		// strip HTML tags
+		$filename = strip_tags( $filename );
+		// replace the surrounding whitespace
+		$filename = trim( $filename, "\t\n\r" );
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: is there a reason for stating those white chars explicitly? please refer to http://php.net/trim

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php [hunk 3]
```diff
@@ -44,15 +48,45 @@ private function getFilePrefixRegex( $contLang ) {
...
+	 * @param $filePrefixRegex
+	 * @return null
+	 */
+	private function extractFilename( $potentialFilename, $filePrefixRegex ) {
+		$trimmedFilename = trim( $potentialFilename, "[]" );
 		$unprefixedFilename = mb_ereg_replace( $filePrefixRegex, "", $trimmedFilename );
-		// strip
 		$filenameParts = explode( '|', $unprefixedFilename );
 		if ( !empty( $filenameParts[0] ) ) {
 			$filename = $filenameParts[0];
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: Do we need this assignment here? Maybe we could just return value here

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php (hunk 1 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: As far as I know, yes. If you have a counterexample, I'm more than willing to factor it in. Aside, this is an additional part of functionality, so we can iterate if the assumption is wrong.

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php (hunk 2 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: Not any more. Good catch!

File: extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php (hunk 3 above)
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/services/Helpers/ImageFilenameSanitizer.php]: I guess one way or the other, no strong preference on my end here

**Comment by idradm** [comment]
it's awesome, :shipit:

File: extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php [hunk 4]
```diff
@@ -10,4 +10,22 @@ static public function onBeforePageDisplay( OutputPage $out, Skin $skin ) {
...
+	 * Store information about raw content of all galleries in article to handle images in infoboxes
+	 *
+	 * @param $name Parser tag name
+	 * @param $marker substitution marker
+	 * @param $content raw tag contents
+	 * @param $attributes
+	 * @param $parser
+	 * @param $frame
+	 */
+	static public function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php]: Nice hook place!

File: extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php [hunk 5]
```diff
@@ -10,4 +10,22 @@ static public function onBeforePageDisplay( OutputPage $out, Skin $skin ) {
...
+		if ( $name === 'gallery' ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php]: Could we const that?

File: extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxDataBag.php [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	public function setGallery( $marker, $content ) {
+		$this->galleries[$marker] = $content;
+	}
+
+	public function getGallery( $marker ) {
+		if(isset($this->galleries[$marker])) {
+			return $this->galleries[$marker];
+		} else {
```
Comment (idradm) [inline extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxDataBag.php]: Do we need this else here?

**Comment by idradm** [comment]
Add const, change ifs formatting in PortableInfoboxDataBag, and it's good to go

**Comment by idradm** [comment]
LG

**Pull request merged**